### PR TITLE
[AArch64]: Added code for generating XAR instruction

### LINKED ISSUE #61584: Patterns to generate RAX1 and XAR on AArch64
LLVM already [generates](https://godbolt.org/z/aKEne9cTj) BCAX and EOR3 from the ARMv8.2-SHA crypto extension set. It doesn't for RAX1 and XAR yet.

### PR DESCRIPTION
**Pull request by BK1603**
cc @davemgreen for review.

Fixes #61584

### CODE REVIEW

File: llvm/lib/Target/AArch64/AArch64ISelDAGToDAG.cpp [hunk 1]
```diff
@@ -4273,6 +4275,41 @@ bool AArch64DAGToDAGISel::trySelectCastScalableToFixedLengthVector(SDNode *N) {
...
+
+  SDValue N0 = N->getOperand(0);
+  SDValue N1 = N->getOperand(1);
+
+  if (N0->getOpcode() != AArch64ISD::VSHL ||
+      N1->getOpcode() != AArch64ISD::VLSHR)
+    return false;
+
+  if (N0->getOperand(0) != N1->getOperand(0) ||
+      N0->getOperand(0)->getOpcode() != N1->getOperand(0)->getOpcode() ||
```
Comment (davemgreen) [inline llvm/lib/Target/AArch64/AArch64ISelDAGToDAG.cpp:4289]: This line wont be needed with the one above.

**Review by davemgreen — COMMENTED** [review]
Thanks for moving this over. This LGTM with the extra line removed.

File: llvm/lib/Target/AArch64/AArch64ISelDAGToDAG.cpp (hunk 1 above)
Comment (BK1603) [reply llvm/lib/Target/AArch64/AArch64ISelDAGToDAG.cpp]: Ah right, missed this. Resolved.

**Review by davemgreen — APPROVED** [review]
Thanks for the patch. This looks like good stuff

**Pull request merged**